code/modules/admin/verbs/secrets.dm

```diff
@@ -348,8 +348,16 @@ GLOBAL_DATUM(everyone_a_traitor, /datum/everyone_is_a_traitor_controller)
 			if(!is_funmin)
 				return
 			SSblackbox.record_feedback("nested tally", "admin_secrets_fun_used", 1, list("Break All Lights"))
-			message_admins("[key_name_admin(holder)] broke all lights")
+			var/station_only = tgui_alert(usr,"Break which lights?", "And the lord said...", list("All Lights", "Station Z Only"))
+			switch(station_only)
+				if("All Lights")
+					station_only = FALSE
+				if("Station Z Only")
+					station_only = TRUE
```

Comment:
if they x out of the alert this should abort

Reply:
```suggestion
				if("All Lights")
					station_only = FALSE
				if("Station Z Only")
					station_only = TRUE
				else
					return
```

Reply:
good call